src/Gaufrette/Adapter/AzureBlobStorage.php

```diff
@@ -47,22 +49,52 @@ class AzureBlobStorage implements Adapter,
     protected $blobProxy;
 
     /**
+     * @var bool
+     */
+    protected $multiContainerMode = false;
+
+    /**
+     * @var CreateContainerOptions
+     */
+    protected $createContainerOptions;
+
+    /**
      * @param AzureBlobStorage\BlobProxyFactoryInterface $blobProxyFactory
-     * @param string                                     $containerName
+     * @param string|null                                $containerName
      * @param bool                                       $create
      * @param bool                                       $detectContentType
+     *
+     * @throws \RuntimeException
      */
-    public function __construct(BlobProxyFactoryInterface $blobProxyFactory, $containerName, $create = false, $detectContentType = true)
+    public function __construct(BlobProxyFactoryInterface $blobProxyFactory, $containerName = null, $create = false, $detectContentType = true)
     {
         $this->blobProxyFactory = $blobProxyFactory;
         $this->containerName = $containerName;
         $this->detectContentType = $detectContentType;
-        if ($create) {
+        if (null === $containerName) {
+            $this->multiContainerMode = true;
+        } elseif ($create) {
             $this->createContainer($containerName);
         }
     }
 
     /**
+     * @return CreateContainerOptions
+     */
+    public function getCreateContainerOptions()
+    {
+        return $this->createContainerOptions;
+    }
+
+    /**
+     * @param CreateContainerOptions $options
+     */
+    public function setCreateContainerOptions(CreateContainerOptions $options)
```

Comment:
I agree actually, did this to keep BC. 👍 

If null, it will be treated as an empty `CreateContainerOptions`, meaning containers get created with whatever the default values other than name are. We can go with this for now and revamp the whole adapter interface with 1.0?